spark-job-server-proposal.md

```diff
@@ -0,0 +1,61 @@
+# Adding Spark Job Server To Run Spark Jobs
+
+## Introduction
+
+This document provides design of spark job server which would trigger spark
+jobs either as scheduled or  batch jobs. Providing the user a ui for inputing
+the jar location, environment variables required to integrate with other apps,
+input parameters for spark job and files required.
+
+## Problem Statement
+
+Spark Applications can only run within OpenShift environment and users are
+required to do provide only source code which currently we use source to image
+to create a standard image which will be run against the spark cluster.
+Currently there is no way to schedule jobs or to run cron jobs. Also if
+users have a more complex setup where they would need infinispan-server or
+other servers they would be required to do a lot of setup.
+
+## Proposed Solution
+
+Apache Spark users will submit their spark job in the form of binary jar or py
+files. These files will require users to provide input as well as be triggered
+by spark-submit shell script. However there are other environment
+requirements that may be needed by users as well.
+
+The requirements to create, run, rerun, stop a particular spark job by one
+click of a button.
```

Comment:
I agree we can keep it stateless and find the current s2i images that are available users can select the one they would like to utilize. Currently this is manual work that adopters of technology would have to do. This would automate the process of templates being used to create pods that run spark job. Rerun can look at already completed spark jobs by redeploying the dc. We should be able to look up previous runs of the dc.